### PR TITLE
WT-6309 Add support for start/stop arguments to wt printlog command (v4.0 backport)

### PR DESCRIPTION
**Pull request by tammybailey**
WT-6309 Add support for start/stop arguments to wt printlog command

* Added optional arguments [-l start-file,start-offset]|[-l start-file,start-offset,end-file,end-offset] to start and stop the printlog at given LSN's. Details included in the printlog usage string.
* Added python tests to test_txn08 to test the start, stop functionality.

(cherry picked from commit e39ffb554160de902060cd063c4b1547ff6d5e1e)
Co-authored-by: Sean Watt <sean.watt@mongodb.com>
(cherry picked from commit 11457083d426362de4528ed9aee024d86f090dc2)

### CODE REVIEW

**Comment by luke-pearson** [comment]
Sean's email seems to have been garbled somewhere along the way.

File: src/log/log.c [hunk 1]
```diff
@@ -2161,16 +2161,17 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
...
+    if (start_lsnp != NULL) {
         /*
          * Offsets must be on allocation boundaries. An invalid LSN from a user should just return
          * WT_NOTFOUND. It is not an error. But if it is from recovery, we expect valid LSNs so give
          * more information about that.
          */
-        if (lsnp->l.offset % allocsize != 0) {
-            if (LF_ISSET(WT_LOGSCAN_RECOVER))
+        if (start_lsnp->l.offset % allocsize != 0) {
+            if (LF_ISSET(WT_LOGSCAN_RECOVER | WT_LOGSCAN_RECOVER_METADATA))
```
Comment (luke-pearson) [inline src/log/log.c:2172]: I am confused by the addition of WT_LOGSCAN_RECOVER_METADATA

Comment (tammybailey) [reply src/log/log.c:2172]: `cherry-pick` pulled it in from 4.2.  Do we want to remove it?

**Comment by tammybailey** [comment]
> Sean's email seems to have been garbled somewhere along the way.

@luke-pearson I'm seeing email addresses like this throughout the git log. I cleaned up the log message in any case.

File: src/log/log.c (hunk 1 above)
Comment (luke-pearson) [reply src/log/log.c:2172]: The original change WT-4908 never had backports requested on it. I'd say remove it?

**Review by Sean04 — APPROVED** [review]
LGTM.

File: src/log/log.c (hunk 1 above)
Comment (tammybailey) [reply src/log/log.c]: Removed.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**